includes/api/class-wc-admin-rest-taxes-controller.php

```diff
@@ -24,4 +24,162 @@ class WC_Admin_REST_Taxes_Controller extends WC_REST_Taxes_Controller {
+
+		$query = "
+			SELECT *
+			FROM {$wpdb->prefix}woocommerce_tax_rates
+			WHERE 1 = 1
+		";
+
+		// Filter by tax class.
+		if ( ! empty( $prepared_args['class'] ) ) {
+			$class  = 'standard' !== $prepared_args['class'] ? sanitize_title( $prepared_args['class'] ) : '';
+			$query .= " AND tax_rate_class = '$class'";
+		}
+
+		/**
+		 * Filter the query string to conditionally return tax codes in the REST API.
+		 *
+		 * @todo Remove this if https://github.com/woocommerce/woocommerce/pull/22813 gets merged.
+		 * @param string $query         Query string used to look up tax codes.
+		 * @param array  $prepared_args Array of arguments for $wpdb->get_results().
+		 */
+		$query = apply_filters( 'woocommerce_rest_tax_query_string', $query, $prepared_args );
+
+		// Order tax rates.
+		$order_by = sprintf( ' ORDER BY %s', sanitize_key( $prepared_args['orderby'] ) );
+
+		// Pagination.
+		$pagination = sprintf( ' LIMIT %d, %d', $prepared_args['offset'], $prepared_args['number'] );
+
+		// Query taxes.
+		$results = $wpdb->get_results( $query . $order_by . $pagination ); // @codingStandardsIgnoreLine.
+
+		$taxes = array();
+		foreach ( $results as $tax ) {
+			$data    = $this->prepare_item_for_response( $tax, $request );
+			$taxes[] = $this->prepare_response_for_collection( $data );
+		}
+
+		$response = rest_ensure_response( $taxes );
+
+		// Store pagination values for headers then unset for count query.
+		$per_page = (int) $prepared_args['number'];
+		$page = ceil( ( ( (int) $prepared_args['offset'] ) / $per_page ) + 1 );
+
+		// Query only for ids.
+		$wpdb->get_results( str_replace( 'SELECT *', 'SELECT tax_rate_id', $query ) ); // @codingStandardsIgnoreLine.
+
+		// Calculate totals.
+		$total_taxes = (int) $wpdb->num_rows;
+		$response->header( 'X-WP-Total', (int) $total_taxes );
+		$max_pages = ceil( $total_taxes / $per_page );
+		$response->header( 'X-WP-TotalPages', (int) $max_pages );
+
+		$base = add_query_arg( $request->get_query_params(), rest_url( sprintf( '/%s/%s', $this->namespace, $this->rest_base ) ) );
+		if ( $page > 1 ) {
+			$prev_page = $page - 1;
+			if ( $prev_page > $max_pages ) {
+				$prev_page = $max_pages;
+			}
+			$prev_link = add_query_arg( 'page', $prev_page, $base );
+			$response->link_header( 'prev', $prev_link );
+		}
+		if ( $max_pages > $page ) {
+			$next_page = $page + 1;
+			$next_link = add_query_arg( 'page', $next_page, $base );
+			$response->link_header( 'next', $next_link );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Add tax code query arguments from request.
+	 *
+	 * @param array $prepared_args Array of arguments used for querying taxes.
+	 * @param array $request Array of request parameters.
+	 * @return array
+	 */
+	public static function add_tax_code_query_args( $prepared_args, $request ) {
+		if ( $request['code'] ) {
+			$prepared_args['code'] = $request['code'];
+		}
+		return $prepared_args;
+	}
+
+	/**
+	 * Filter tax codes by code.
+	 *
+	 * @param string $query Sql query string.
```

Comment:
Nit-picking here, but could we capitalize SQL?
```suggestion
	 * @param string $query SQL query string.
```